add generate_expert_indices op

shanjiang7 opened the pull request:
<!-- Demo: https://github.com/PaddlePaddle/PaddleNLP/pull/26 -->
#### Before submitting

- [ ] Lint code. If there are lint issues, please format the code first.

```shell
# Install and register `pre-commit` in the project folder
pip install pre-commit && pre-commit install

# Process previous code files separately
pre-commit run --file XXXX.py
```

- [ ] Add test cases into `tests` folder. If there are codecov issues, please add tests cases first.

### PR types
<!-- One of [ New features | Bug fixes | Function optimization | Performance optimization | Breaking changes | Others ] -->

### PR changes
<!-- One of [ Models | APIs | Docs | Others ] -->

### Description
<!-- Describe what this PR does -->

Commit:
fused_indices

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/PaddlePaddle/PaddleNLP?pullRequest=10705) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/PaddlePaddle/PaddleNLP?pullRequest=10705) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: shanjiang7<br/>:x: jiangshan18<br/><hr/>**jiangshan18** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/PaddlePaddle/PaddleNLP?pullRequest=10705) it.</sub>

shanjiang7 committed:
update

 Changes to be committed:
	modified:   slm/model_zoo/gpt-3/external_ops/fused_quanted_ops/fused_indices.cu
	modified:   slm/model_zoo/gpt-3/external_ops/setup_fp8.py
	new file:   tests/ops/test_fused_indices.py

 Untracked files:
	slm/model_zoo/gpt-3/external_ops/fused_quanted_ops/fused_indices_1.cu
	slm/model_zoo/gpt-3/external_ops/test.py
